### DISCUSSION TITLE
Libraries management - Oct '23 meeting

### DISCUSSION THREAD

**Discussion by woodseowl** [CornellCustomDev/CD-LaravelStarterKit#53] [General]
Notes from Oct. 16, 2023 meeting. Agenda:

> Purpose: Review the idea of building out libraries in the Laravel Starter Kit instead of as separate packages (as a follow-up to the Slack discussion). Discuss other ideas / questions about work on the Starter Kit.
>
> A primary goal of this meeting will be to verify that the approach to adding the [PhoneNumber library](https://github.com/CU-CommunityApps/CD-LaravelStarterKit/pull/52) is a direction that makes sense to people in general. If so, that draft PR will be ready for review.
>
> Additionally, what ideas or questions do people have about contributing to the Starter Kit?

Notes and actions from the meeting:

- We planned to discuss the agenda (above), along with possibly discussing how we can add new approaches like Livewire 3 + Volt

- Libraries versus separate packages:
  - General consensus that adding internal packages (possibly each in their own directory) is better than multiple packages in separate repositories.
  - It is possible to use the Starter Kit simply as a package library (just skip the installation steps), so if all you want is to use one of the packages on an existing project, that is possible.
  - If we later find that we are getting a lot of bulk in the Starter Kit, we should be able to move components to individual packages and manage those dependencies internally.

- Feedback on contributing to the Starter Kit
  - Not having separate packages for each library will help it be shared code instead of silos of ownership.
  - How can we maintain it?
    - If the contributed work is complex, what is the cost of maintaining it and how do we keep up with it?
      - Eric's contributions look hard to maintain because they require a lot of work created because of test-driven development.
    - How do we keep it from becoming a dead project? (Meaning a few years from now it is no longer used.)
      - Like the Drupal Starter Kit, if we create a lot of value for a new project it will naturally continue to be used.
  - We should put in work that we would really benefit from (e.g., common form inputs)
    - It is better to get in good examples as they are than have things need to pass a high bar of quality standards.

**Outcomes and Actions**
- Eric will refactor the LDAP library example to follow the feedback above
- Input form components
  - Eric will set up a place for discussion / comparison
  - Rich and Nazrin will contribute potential components from existing projects
    - Possibly using Livewire 3 + Volt
  - Laravel Demo project can be used to test the components: https://github.com/CU-CommunityApps/CD-laravel-demo

**Comment by inaydich** [comment]
Looks good. Thank you for bringing this discussion and creating these notes!  I am looking forward to contributing to the project.
